_posts/2024-06-14-devlog-28.md

```diff
@@ -0,0 +1,131 @@
+
+So I made a butterfly:
+
+<video width="100%" autoplay="autoplay" loop="true" muted>
+  <source src="https://i.imgur.com/cNxPs8h.mp4" type="video/mp4" />
+</video>
+
+I figured something small like a butterfly would be a great first step in learning some basics about game dev, like how to make models, materials and apply those to game objects. Dipping my toes in ended up turning into a deep dive on shadergraph, a quick bootcamp on how things are connected in Unity, re-learning some algebra, and watching dozens of butterfly videos on youtube. 
+
+With guidance from Scott, I started on a simple plan:
+1. Make a flat quad model that represented half of a butterfly
+2. Use a shader to cut out a butterfly shape from the quad
+3. Use that same shader to mirror that quad in texture and angle to create the butterfly's other half
+4. Move that quad around to mimic butterfly behavior
+
+After getting the model and textures I needed into the game, a quick seminar on DOTS and ECS from Scott got me up to speed on the code I would need to write in order to get my butterfly fluttering. With some help from the fantastic graphing calculator website Desmos, I was able to figure out an equation for calculating the path the butterflies would take from one landing zone to the next. Add in some code to oscillate the butterfly's position along that path, control how fast and to what degree they move their wings, pick new targets once they reach their destination, and to relax a bit before taking off and voila, you've got yourself a decent looking butterfly!
+
+{% include gallery id="gallery2" layout="half" caption="Butterfly Development Artifacts" %}
+
+Now.. how are these things going to spawn?
+
+## Time of Day
+
+I'm a lucky guy; I live near one of the largest public parks in Seattle, one with a gorgeous view across the Puget Sound where you can watch the sun set behind the Olympic Mountains as it shines brilliantly off the water. I regularly go for walks through the park and one thing that always strikes me is how different the park can feel depending on the time of day. In the morning, robins and jays call out non-stop, the squirrels scurry to and fro looking for snacks, all while the trees and ferns perk up to bask in the sun. At dusk, rabbits emerge from their dens to forage while the crows stopp their murderous cawing, tucking into their nests for the night. There's an entire ecosystem going about its day-to-day without a care in the world for the dog-walkers and bird-watchers. That's the kind of feeling I also want to capture in Automation Station when it comes to the flora and fauna. They'll just keep going about their business unless you bother them.
```

Comment:
Typo in "crows stopp their murderous cawing". Unless that's a word I've never heard of before haha.